### DISCUSSION TITLE
Vim to Meow: Muscle Memory

### DISCUSSION THREAD

**Discussion by willbush** [meow-edit/meow#628] [Q&A] [answered]
I've migrated from qwerty to colemak mod-DH, in the past, and it overwrote my muscle memory completely. Only after the fact did I realize other people practice both as they go and can switch between the two layouts.

So I've been wondering what it's like to have migrated from vim / evil to meow. Should I practice both?

A lot of editors have vim emulation which is a pretty nice-to-have feature if I need to use something else. Wondering if its possible to approximate meow basics with a vim config, if I should try to keep vim muscle memory, or just say
screw it and switch?

**Comment by okamsn** [comment, marked as answer]
As a former Evil user who was already familiar with the basic Emacs key bindings, I didn't find Meow to be a big jump. There was a small period of time getting used to the selection-first behavior, but now I see that behavior as just a faster way of doing what one would do in any other editor. I don't feel any limitation from switching to Meow, since it still provides text-object-like convenience via ["thing-based selection"](https://github.com/meow-edit/meow/blob/master/COMMANDS.org#thing-based-selection).

On any difficulties from possibly switching between the two frequently, I don't have an opinion as I haven't tried that.

**Comment by eshrh** [comment]
When I switched from qwerty to dvorak I also practically lost the ability to type qwerty at a reasonable speed. I think that keyboards are probably a bigger problem for me than editors... I might sometimes have to use other computers for a short while (which means using the keyboard), but i'll never sit down to edit a file there without setting up the keymap first. 

It's really not hard to move your emacs config wherever you need it, especially if you use a purely functional package manager like straight/nix/guix. I do (somewhat) maintain an "init_minimal.el" which contains my bare essentials, which in terms of external packages is just:
- https://github.com/eshrh/nyaatouch (avy, swiper, smartparens)
- vertico

At least the default meow keybindings let you keep vim-style hjkl movement keys. I was already screwed with other editors the moment I switched keyboards unless I stuck to using hjkl which I honestly think is a bit miserable. Meow makes things so much nicer because it's so easy to make your own totally custom setup.
